File: CSC454HW6/CSC454HW6/src/csc454hw6/Mem.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package csc454hw6;

/**
 *
 * @author chrisrk192
 */
public class Mem implements ModelInterface {
    int qHead;
    int qTail;
    
    @Override
    public String GetInitilizationSignature() {
        return "initialState1, initialState2";
    }

    @Override
    public void Initialize(String[] args) {
        if(args.length >= 2){
            qHead = Integer.parseInt(args[0]);
            qTail = Integer.parseInt(args[1]);
        }
        else
            qHead = 0; qTail = 0;
    }
    
    @Override
    public String[] Tick(String[] args) throws CantPerformAction{
        String[] out = OutPut();
        StateTransition(args);
        return out;
    }
    
    @Override
    public String[] OutPut() {
        
        String[] s = new String[1];
        s[0] = qHead + "";
        return s;
    }

    @Override
    public void StateTransition(String[] inputItems) throws CantPerformAction {
        
        qHead = qTail;
        qTail = Integer.parseInt(inputItems[0]);
    }

    @Override
    public String GetStatus() {
        return "Queue: " + qHead + ", " + qTail;
    }

    @Override
    public Object Clone() {
        Mem m = new Mem();
        m.qHead = this.qHead;
        m.qTail = this.qTail;
        return m;
    }
    
}

File: CSC454HW5/CppImpOfHW4(oldNoWork)/Mem.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   Mem.cpp
 * Author: chrisrk192
 * 
 * Created on October 27, 2017, 10:24 PM
 */

#include "Mem.h"

Mem::Mem() {
}

Mem::Mem(const Mem& orig) {
}

Mem::~Mem() {
}

std::string Mem::GetInitilizationSignature(){
    return "initialState1, initialState2";
}
void Mem::Initialize(std::vector<std::string>args){
    if(args.size() >= 2){
            qHead = std::stoi(args.at(0));
            qTail = std::stoi(args.at(1));
        }
        else
            qHead = 0; qTail = 0;
}
std::vector<std::string> Mem::OutPut(){
    std::vector<std::string> o;
    o.push_back(std::to_string(qHead));
    return o;
}  
void Mem::StateTransition(std::vector<std::string>args){
    qHead = qTail;
    qTail = std::stoi(args.at(0));
}
std::string Mem::GetStatus(){
    return "Queue: " + std::to_string(qHead) + ", " + std::to_string(qTail);
}
std::vector<std::string> Mem::Tick(std::vector<std::string> inputs){
    std::vector<std::string> out = OutPut();
    StateTransition(inputs);
    return out;
}